kyo-pure/shared/src/main/scala/kyo/core.scala

```diff
@@ -0,0 +1,214 @@
+package kyo
+
+import kyo.Locals.State
+import scala.annotation.tailrec
+
+object core:
+
+    import internal.*
+
+    type Id[T]    = T
+    type Const[T] = [U] =>> T
+
+    opaque type <[+T, -S] >: T = T | internal.Kyo[T, S]
+
+    trait Effect[+E]:
+        type Command[_]
+
+    extension [E <: Effect[E]](e: E)
+
+        inline def suspend[T](inline cmd: e.Command[T])(
+            using inline _tag: Tag[E]
+        ): T < E =
+            new Suspend[e.Command, T, T, E]:
+                def command = cmd
+                def tag     = _tag.asInstanceOf[Tag[Any]]
+                def apply(v: T, s: Safepoint[E], l: State) =
+                    v
+
+        inline def suspend[T, U, S](inline cmd: e.Command[T], inline f: T => U < S)(
+            using inline _tag: Tag[E]
+        ): U < (E & S) =
+            new Suspend[e.Command, T, U, S]:
+                def command = cmd
+                def tag     = _tag.asInstanceOf[Tag[Any]]
+                def apply(v: T, s: Safepoint[S], l: State) =
+                    f(v)
+    end extension
+
+    inline def eval[T, S, S2](v: T < S)(inline f: (Safepoint[S & S2], () => T < S) => T < (S & S2)): T < (S & S2) =
```

Comment:
This is a new method I introduced to enable implementing `IOs` methods like `handle` and `ensure` without depending on internal core classes.